legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/platformBinding/typeInfo/typeInfo.pure

```diff
@@ -649,3 +656,89 @@ function  <<access.private>> meta::pure::executionPlan::platformBinding::typeInf
 {
    {pe1:PackageableElement[1], pe2:PackageableElement[1] | $pe1->elementToPath() == $pe2->elementToPath()};
 }
+
+function <<access.private>> meta::pure::executionPlan::platformBinding::typeInfo::typeInfoFromTreeExtra(info:TypeInfoSet[1],tree:GraphFetchTree[1], debug:DebugContext[1]): TypeInfoSet[1]
+{
+  print(if($debug.debug,|$debug.space+'getting extra types from tree : '+  + $tree->asString(false) + '\n',|''));
+                                    
+  let extraTypes= $tree->match([
+    p:RoutedPropertyGraphFetchTree[1]| 
+             $p.propertyMapping->map(pm|  
+                  $pm->match([
+                        ppm: meta::pure::mapping::modelToModel::PurePropertyMapping[1] | 
+                            $ppm.transform->evaluateAndDeactivate()->scanTypesFromAny([], $debug->indent()->indent());,
+                        apm: Any[1] | []
+                    ]));,
+    r:RootGraphFetchTree<Any>[1] | []
+  ])->remove(Any)->remove(Function)->removeDuplicates(elementsEquals())->cast(@Type);
+  print(if($debug.debug,|$debug->indent().space+'found extraTypes: '+ $extraTypes->map(t|$t->toString())->makeString('[', ', ', ']') + '\n',|''));
+  
+  let withExtraTypes = $extraTypes->fold({t, tis| $tis->addBasicTypeInfoIfMissing($t)}, $info);
+   
+   // TODO restrict the properties added for class,
```

Comment:
If all properties are being added, can it potentially explode the graph? Should we address this TODO

Reply:
I have restricted only simple/complex properties, no derived /association properties. Also these are added for new class types only . Can have further look after lambda tests?